### DISCUSSION TITLE
REQ: Streamer mode ('asterix to hide password in /password and /spectator password')

### DISCUSSION THREAD

**Discussion by rakasunka** [ezQuake/ezquake-source#495] [Ideas]
**Is your feature request related to a problem? Please describe.**
Would be nice to hide password in /password and /spectator.

**Describe the solution you'd like**
Se above

**Describe alternatives you've considered**
No idea.

**Additional context**
Streaming game from a password protected server without giving out password by accident.

**Impact on server or mods**

**Comment by Calinou** [comment]
I doubt this can be done given how the console works. Instead, ezQuake should be able to use a separate variable (loaded into `autoexec.cfg` or similar) or even a system environment variable as a password (`EZQUAKE_PASSWORD`, `EZQUAKE_SPECTATOR_PASSWORD`).

**Comment by meag** [comment]
(other people have got round this before using aliases in a config file)

I think you're asking for some kind of streaming mode?  We could try and filter characters if certain strings have appeared before them, but it wouldn't be guaranteed to work.  Unless the console was in a different window to the playing window, but that would be a big change.

**Comment by rakasunka** [comment]
Yeah filter it out after a certain string is exactly what i had in mind. But you're right. Putting it in alias/config is a probably a better idea

**Comment by jite** [comment]
Another suggestion would be to have `/password` act like a command instead, so hit /password [ENTER] and then provide an input box (just echo `Password: ` in the console) that doesn't echo what's being typed, not even * (to avoid giving away number of chars).

**Comment by interacca** [comment]
Capisco perfettamente il problema: quando si streamma un gioco da un server protetto da password, c’è il rischio di mostrarla accidentalmente a tutti. Sarebbe molto utile avere una funzione che nasconda automaticamente [bet-on-red.it](https://bet-on-red.it/) la password nei comandi come /password e /spectator, così da poter trasmettere senza preoccupazioni. Non vedo alternative semplici a questa soluzione, perché altrimenti bisognerebbe ricordarsi di non digitare mai la password in pubblico. Implementare questa funzione renderebbe lo streaming più sicuro e protegge il server da accessi non autorizzati.

**Comment by JesseReed123** [comment]
Bilo bi zelo uporabno, če bi bilo mogoče skriti geslo pri ukazih /password in /spectator. To bi preprečilo, da se med streamanjem igre s strežnikom, zaščitenim z geslom, geslo po nesreči razkrije. Idealna rešitev bi bila, da https://roobetslovene.com/ vneseno geslo ne bi bilo vidno na zaslonu ali v konzoli, medtem ko se še vedno uporablja za prijavo. Alternativnih rešitev trenutno nimam, vendar bi to izboljšalo varnost in preprečilo nenamerne razkritja gesel med prenosom v živo.
